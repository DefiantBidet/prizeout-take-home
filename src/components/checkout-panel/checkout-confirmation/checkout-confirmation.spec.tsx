import React from 'react';
import '@testing-library/jest-dom';

import { render, screen } from 'Testing/test-utils';

import CheckoutConfirmationPanelView from './checkout-confirmation';

describe('Test Checkout Confirmation Component', () => {
    test('Component matches snapshot', () => {
        const { asFragment } = render(<CheckoutConfirmationPanelView />);

        expect(asFragment()).toMatchSnapshot();
    });

    test('Component renders header copy', () => {
        render(<CheckoutConfirmationPanelView />);

        expect(screen.queryByText('Checkout Confirmation Panel')).toBeInTheDocument();
    });

    test('Component renders back button', () => {
        render(<CheckoutConfirmationPanelView />);

        expect(screen.getByRole('button')).toBeInTheDocument();
    });
});
