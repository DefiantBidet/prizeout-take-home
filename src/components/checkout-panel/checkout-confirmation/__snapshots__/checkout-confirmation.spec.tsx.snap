// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Checkout Confirmation Component Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="checkout-panel__view"
  >
    <section
      class="checkout-confirmation"
    >
      <h2>
        Checkout Confirmation Panel
      </h2>
      <div>
        confirmation: 
      </div>
      <button>
        Done
      </button>
    </section>
  </div>
</DocumentFragment>
`;
